Source/source/GPS.cpp

```diff
@@ -60,6 +62,41 @@ string toStringTime(struct minmea_time *time) {
 	return timeString.str();
 }
 
+
+unsigned int encode_time_as_int(struct minmea_time *time) {
+    unsigned int BITS = 32;
+    unsigned int H_BITS = 5;
+    unsigned int M_BITS = 6;
+    unsigned int S_BITS = 6;
+    unsigned int enc = 0;
+    unsigned int hours = (unsigned int)(time->hours);
+    unsigned int minutes = (unsigned int)(time->minutes);
+    unsigned int seconds = (unsigned int)(time->seconds);
+    
+    enc = enc | hours << BITS - H_BITS >> BITS - H_BITS - M_BITS - S_BITS;
+    enc = enc | minutes << BITS - M_BITS >> BITS - M_BITS - S_BITS;
+    enc = enc | seconds << BITS - S_BITS >> BITS - S_BITS;
```

Comment:
Order of operation is _just_ enough to let this work (see [Operator Precedence](https://en.cppreference.com/w/cpp/language/operator_precedence)). For code clarity, I'd recommend adding parentheses here.